python/pyspark/rdd.py

```diff
@@ -188,6 +188,12 @@ def __init__(self, jrdd, ctx, jrdd_deserializer=AutoBatchedSerializer(PickleSeri
         self._id = jrdd.id()
         self.partitioner = None
 
+    def __enter__(self):
```

Comment:
Is it reasonable just to raise an error saying that the context manager is meant to work only with cached RDD's (Dataframes) if `self.is_cached` is not set to True?